engine/access/rpc/backend/backend_stream_transactions.go

```diff
@@ -241,43 +241,40 @@ func (b *backendSubscribeTransactions) searchForTransactionBlockInfo(
 	}
 
 	if collectionID != flow.ZeroID {
-		return block.Header, block.ID(), height, collectionID, nil
+		return block, block.ID(), height, collectionID, nil
 	}
 
 	return nil, flow.ZeroID, 0, flow.ZeroID, nil
 }
 
-// searchForTransactionResult searches for the transaction result of a block. It retrieves the execution result for the specified block ID.
-// Expected errors:
-// - codes.Internal if an internal error occurs while retrieving execution result.
+// searchForTransactionResult searches for the transaction result of a block. It retrieves the transaction result from
+// storage and, in case of failure, attempts to fetch the transaction result directly from the execution node.
+// This is necessary to ensure data availability despite sync storage latency.
+//
+// No errors expected during normal operations.
 func (b *backendSubscribeTransactions) searchForTransactionResult(
 	ctx context.Context,
 	txInfo *TransactionSubscriptionMetadata,
 ) (*access.TransactionResult, error) {
-	_, err := b.executionResults.ByBlockID(txInfo.BlockID)
+	txResult, err := b.backendTransactions.GetTransactionResultFromStorage(ctx, txInfo.blockWithTx, txInfo.TransactionID, txInfo.eventEncodingVersion)
 	if err != nil {
-		if errors.Is(err, storage.ErrNotFound) {
-			return nil, nil
-		}
-		return nil, fmt.Errorf("failed to get execution result for block %s: %w", txInfo.BlockID, err)
-	}
-
-	txResult, err := b.backendTransactions.GetTransactionResult(
-		ctx,
-		txInfo.TransactionID,
-		txInfo.BlockID,
-		txInfo.CollectionID,
-		txInfo.eventEncodingVersion,
-	)
+		// If any error occurs with local storage - request transaction result from EN
+		txResult, err = b.backendTransactions.GetTransactionResultFromExecutionNode(
+			ctx,
+			txInfo.blockWithTx,
+			txInfo.TransactionID,
+			txInfo.eventEncodingVersion,
+		)
 
-	if err != nil {
-		// if either the storage or execution node reported no results or there were not enough execution results
-		if status.Code(err) == codes.NotFound {
-			// No result yet, indicate that it has not been executed
-			return nil, nil
+		if err != nil {
+			// if either the execution node reported no results
+			if status.Code(err) == codes.NotFound {
+				// No result yet, indicate that it has not been executed
+				return nil, nil
```

Comment:
This convention is not explained in the documentation, usually if there is no error then a result has to be available, returning `nil, nil` is undesirable, I am leaning to updating the documentation and returning a sentinel error to inform caller that there is no data available.